### PR TITLE
Feature/more stream tests

### PR DESCRIPTION
**Pull request by gbrova**
@tmgordeeva, I added a few more tests to `StreamTest`.  Could you take a quick look and let me know if this is on the right track?

Add some more verification to StreamTest

- Port some tests (`testStreamMetadataConflict...`, `testStoreCopy`, `testLookupStreamIdsByHash`)  from MetropolisStreamingTest. 
- Cover all loadStream functions (`loadStream`, `loadStreams`, `loadStreamAsFile`).
- Test both `storeStream` and `storeStreams`.

### CODE REVIEW

File: atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/StreamTest.java [hunk 1]
```diff
@@ -164,20 +183,189 @@ private long storeAndCheckExpiringByteStreams(int size) throws IOException {
...
+
+    private void assertStreamHasBytes(InputStream stream, byte[] bytes) throws IOException {
+        byte[] streamAsBytes = IOUtils.toByteArray(stream);
+        Assert.assertArrayEquals(bytes, streamAsBytes);
+    }
+
+    @Test
+    public void testLookupStreamIdsByHash() throws Exception {
+        final byte[] bytes1 = new byte[2 * StreamTestStreamStore.BLOCK_SIZE_IN_BYTES];
+        final byte[] bytes2 = new byte[2 * StreamTestStreamStore.BLOCK_SIZE_IN_BYTES];
```
Comment (schlosna) [inline atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/StreamTest.java]: We should be testing with a variety of input sizes especially on edge cases and bounds of blocks

File: atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/StreamTest.java [hunk 2]
```diff
@@ -164,20 +183,189 @@ private long storeAndCheckExpiringByteStreams(int size) throws IOException {
...
+        Map<Sha256Hash, Long> sha256HashLongMap = txManager.runTaskWithRetry(t -> store.lookupStreamIdsByHash(t, ImmutableSet.of(hash1, hash2, hash3)));
+
+        assertEquals(id1, sha256HashLongMap.get(hash1).longValue());
+        assertEquals(id2, sha256HashLongMap.get(hash2).longValue());
+        assertEquals(null, sha256HashLongMap.get(hash3));
+    }
+
+    @Test
+    public void testStoreCopy() {
+        final byte[] bytes = new byte[2 * StreamTestStreamStore.BLOCK_SIZE_IN_BYTES];
```
Comment (schlosna) [inline atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/StreamTest.java]: Test with multiple size permutations & -1, +1 on bounds

File: atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/StreamTest.java [hunk 3]
```diff
@@ -164,20 +183,189 @@ private long storeAndCheckExpiringByteStreams(int size) throws IOException {
...
+        });
+
+        firstLatch.await();
+
+        Future<?> secondFuture = exec.submit((Runnable) () -> txManager.runTaskThrowOnConflict((TransactionTask<Void, RuntimeException>) t -> {
+            twoConflictingTasks.startSecondAndFinish(t, streamId);
+            return null;
+        }));
+
+        exec.shutdown();
```
Comment (schlosna) [inline atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/StreamTest.java]: I think you want to also shutdown in finally block in case this test fails to avoid flaking others

File: atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/generated/DeletingStreamStore.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+public class DeletingStreamStore {
+    private final StreamTestStreamStore underlyingStreamStore;
+
+    public DeletingStreamStore(StreamTestStreamStore underlyingStreamStore) {
+        this.underlyingStreamStore = underlyingStreamStore;
+    }
+
+    public void deleteStreams(Transaction t, Set<Long> streamIds) {
+        underlyingStreamStore.deleteStreams(t, streamIds);
```
Comment (tmgordeeva) [inline atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/generated/DeletingStreamStore.java]: I don't really get the point of this class if it's just to class this one method on the original stream store.

**Comment by tmgordeeva** [comment]
More edge cases:
- Cleanup tasks (StreamTestIndexCleanupTask, StreamTestMetadataCleanupTask), either run directly or indirectly through scrub/sweep. You can set scrub to run by using setTransactionType to AGGRESSIVE_HARD_DELETE on the transaction.
- Test consistency of length/hash values in stream metadata entries

File: atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/StreamTest.java [hunk 5]
```diff
@@ -164,20 +183,189 @@ private long storeAndCheckExpiringByteStreams(int size) throws IOException {
...
+
+    @Test
+    public void testLookupStreamIdsByHash() throws Exception {
+        final byte[] bytes1 = new byte[2 * StreamTestStreamStore.BLOCK_SIZE_IN_BYTES];
+        final byte[] bytes2 = new byte[2 * StreamTestStreamStore.BLOCK_SIZE_IN_BYTES];
+
+        long id1 = timestampService.getFreshTimestamp();
+        long id2 = timestampService.getFreshTimestamp();
+
+        Random rand = new Random();
```
Comment (schlosna) [inline atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/StreamTest.java]: We probably want to incorporate some utilities around repeatable randomized testing such that we know what the initial seed was so that we can more deterministically reproduce failures. For example, see https://www.elastic.co/blog/elasticsearch-testing-qa-increasing-coverage-randomizing-test-runs and https://lucene.apache.org/core/4_0_0/test-framework/org/apache/lucene/util/LuceneTestCase.html for examples from ES and Lucene

File: atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/generated/DeletingStreamStore.java (hunk 4 above)
Comment (gbrova) [inline atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/generated/DeletingStreamStore.java]: `StreamTestStreamStore.deleteStreams` is package private, this exposes it. Let me know if you have other thoughts about how to change the package structure.

File: atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/StreamTest.java (hunk 5 above)
Comment (tmgordeeva) [inline atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/StreamTest.java]: This sounds like a good idea, but I wouldn't want to block this PR on getting it in.

Comment (schlosna) [inline atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/StreamTest.java]: Works for me
